### PR TITLE
Updates glossary with many terms and links.

### PR DESCRIPTION
**Pull request by KipHamiltons**
Keen to hear feedback and suggestions.
Feel free to suggest more things to define too.
[Preview](https://deploy-preview-79--nubook.netlify.app/guides/general/glossary)

### CODE REVIEW

**Review by ysims — COMMENTED** [review]
Others...

- Simscape: Simulation of the NUgus robot made in Simulink
- Bonn/Nimbro: Nimbro is the multi time champion of the adult size RoboCup competition. They are from the University of Bonn, in Germany. We often just call them 'Bonn'. 
- Machine learning/deep learning
- ZMP maybe?

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 1]
```diff
@@ -5,3 +5,133 @@ title: Glossary
...
+
+[**Kalman Filter**](https://en.wikipedia.org/wiki/Kalman_filter): Control theoretic technique for estimating variables by estimating a joint probability distribution over the variables for each timeframe.
+
+[**Kinematics**](https://en.wikipedia.org/wiki/Kinematics): Mathematics which describes motion, which is used to calculate trajectories and servo outputs to control robots.
+
+[**Lambda Expression**](https://en.cppreference.com/w/cpp/language/lambda): A way of defining an anonymous function in C++.
+
+[**Localisation**](https://en.wikipedia.org/wiki/Robot_navigation): The system the robot uses to work out its position relative to its environment and objects within it.
+
+[**Odometry**](https://en.wikipedia.org/wiki/Odometry): Estimation of change in position based on sensor commands and data.
```
[the remark this answers is not in the document]
Comment (KipHamiltons) [reply src/book/03-guides/04-general/03-glossary.mdx:69]: Don't we take sensor outputs into account for odometry? I had assumed you would

**Comment by ysims** [comment]
I think also add 

- Open Loop: same meaning as static motion, but more general... not just motion
- Closed Loop: as above but for dynamic

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 2]
```diff
@@ -5,3 +5,181 @@ title: Glossary
...
+
+[**NUgan**](https://github.com/NUbots/NUgan): A custom [cycle-gan](https://junyanz.github.io/CycleGAN/) implementation to generate more realistic images for training the vision networks.
+
+[**NUgus**](https://nubook.nubots.net/system/hardware/overview#igus-modifications): NUbots robots as currently used in the RoboCup humanoid league. They are modified [igus Humanoid Open Platform](https://arxiv.org/abs/1809.11110) robots, which we 3D-printed.
+
+[**NUpbr**](https://github.com/NUbots/NUpbr): The software NUbots uses to generate Physically Based Rendered football fields, used to train vision networks.
+
+[**NUsight**](https://github.com/NUbots/NUsight2): NUbots' visualisation software, ran through a browser on an external computer.
+
+**NUsense**: A secret project for sensor IO.
```
Comment (ysims) [inline src/book/03-guides/04-general/03-glossary.mdx:53]:
```suggestion
**NUSense**: A secret project for sensor IO.
```

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 3]
```diff
@@ -5,3 +5,181 @@ title: Glossary
...
+
+**Quintic Walk**: Open loop walk engine which uses quintic splines to create trajectories. Created by Bit-Bots, based off code from team Rhoban, then ported to run with NUClear.
+
+**ScriptTuner**: A NUbots program for manually creating motion targets for movement scripts.
+
+**ScriptRunner**: The program which runs movement scripts on the robot, such as `Zombie.yaml` which is used to test servo installations.
+
+**Static Motion**: We tend to refer to motion which is fully controlled by scripts which do not change at run-time as _Static Motion_.
+
+**Subsumption**: TODO!
```
Comment (ysims) [inline src/book/03-guides/04-general/03-glossary.mdx:65]:
```suggestion
**Subsumption**: The priority system that determines if a neutron gets access to the servos it wants to control.
```
Might need rewording. But something like this.

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 4]
```diff
@@ -5,3 +5,181 @@ title: Glossary
...
+
+[**VisualMesh**](https://arxiv.org/abs/1807.08405): A computer vision technique NUbots uses which overlays a mesh over the image input to more efficiently sample it and look for objects. Source can be found [here](https://github.com/Fastcode/VisualMesh).
+
+### General
+
+[**Affine Transform**](https://en.wikipedia.org/wiki/Affine_transformation): A transform which keeps straight lines straight, and parallel lines parallel, without preserving distances or angles between lines otherwise.
+
+[**Closed Loop Control**](https://en.wikipedia.org/wiki/Closed-loop_transfer_function): Control which uses output from the thing being used as input for controlling it.
```
Comment (ysims) [inline src/book/03-guides/04-general/03-glossary.mdx:73]:
```suggestion
[**Closed Loop Control**](https://en.wikipedia.org/wiki/Closed-loop_transfer_function): Control action which uses output from the system as input.
```

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 5]
```diff
@@ -5,3 +5,181 @@ title: Glossary
...
+
+[**Machine Learning**](https://en.wikipedia.org/wiki/Machine_learning): Systems and models which improve their outputs by exposure to inputs.
+
+[**Mahony Filter**](https://nitinjsanket.github.io/tutorials/attitudeest/mahony): A complementary filter which estimates attitude, angle, and orientation using IMU data.
+
+[**Motion Capture (mocap)**](https://en.wikipedia.org/wiki/Motion_capture): The recording of movements of moving objects, which keeps track of specific points on the object as it moves.
+
+[**Odometry**](https://en.wikipedia.org/wiki/Odometry): Estimation of change in position and orientation based on sensor inputs.
+
+[**Open Loop Controller**](https://en.wikipedia.org/wiki/Open-loop_controller): A controller whose actions aren't affected by feedback from its actions.
```
Comment (ysims) [inline src/book/03-guides/04-general/03-glossary.mdx:111]:
```suggestion
[**Open Loop Controller**](https://en.wikipedia.org/wiki/Open-loop_controller): Control action that does not use output from the system as input.
```

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 6]
```diff
@@ -5,3 +5,181 @@ title: Glossary
...
+
+[**RoboCup**](https://www.robocup.org/): The robotic football world cup, held each year in different cities around the world.
+
+[**RoboCup Symposium**](https://www.robocup.org/symposium): The research conference which is held after RoboCup each year, featuring research from the teams competing.
+
+[**Robotis**](https://www.robotis.us/): The Korean robotics manufacturer which makes the servos that NUgus uses.
+
+[**Simulink**](https://www.mathworks.com/products/simulink.html): A MATLAB-based graphical simulation software.
+
+[**Simscape**](https://www.mathworks.com/products/simscape.html): A MATLAB-based simulation tool for creating models of physical systems in the Simulink environment.
```
Comment (ysims) [inline src/book/03-guides/04-general/03-glossary.mdx:183]:
```suggestion
[**Simscape**](https://www.mathworks.com/products/simscape.html): A MATLAB-based simulation tool for creating models of physical systems in the Simulink environment. Also the name of the GitHub repository that contains the solution that models and simulates the NUgus robot using Simulink and Simscape.
```

File: src/book/03-guides/04-general/03-glossary.mdx [hunk 7]
```diff
@@ -5,3 +5,181 @@ title: Glossary
...
+
+[**Dynamixel**](https://www.robotis.us/dynamixel/): The _Robotis_ servo models which NUgus uses.
+
+[**igus**](https://www.igus.com.au/): The creators of the robotics platform NUgus is based on.
+
+[**igus Humanoid Open Platform**](https://arxiv.org/abs/1809.11110): The 3d printed base design which the NUgus is based on.
+
+[**Eigen**](https://eigen.tuxfamily.org/): The C++ linear algebra library NUbots uses. Documentation can be found [here](https://eigen.tuxfamily.org/dox/index.html).
+
+[**Gazebo**](https://gazebosim.org/): A simulation software used which was used to optimise the "OldWalkEngine" with a genetic algorithm.
```
Comment (ysims) [inline src/book/03-guides/04-general/03-glossary.mdx:157]:
```suggestion
[**Gazebo**](https://gazebosim.org/): A simulation software that is used to optimise motions with a genetic algorithm.
```

File: src/book/03-guides/04-general/03-glossary.mdx (hunk 3 above)
Comment (JosephusPaye) [reply src/book/03-guides/04-general/03-glossary.mdx:65]: We should probably briefly describe what a neutron is here in parenthesis, or add a new term in the glossary and link to that, or link to a definition somewhere in the NUClear docs.

File: src/book/03-guides/04-general/03-glossary.mdx (hunk 7 above)
Comment (KipHamiltons) [reply src/book/03-guides/04-general/03-glossary.mdx:157]: I don't agree with this one. This makes it seem like that's the point of gazebo, but gazebo is a lot more than just that project - especially this year with RoboCup

**Comment by JosephusPaye** [comment]
I've been thinking about how to design this page ever since we started NUbook. How's this look?

**Comment by KipHamiltons** [comment]
@JosephusPaye I changed that word btw ;P

**Review by JosephusPaye — APPROVED** [review]
LGTM. I'll add the new components for styling later and we can update it then.

As a reminder for future me, we want to be able to mark terms as internal or external, with appropriate links.

**Pull request merged**